### PR TITLE
[New Rule] Windows Service Installed via an Unusual Client

### PR DESCRIPTION
**Pull request by Samirbous**
https://www.x86matthew.com/view_post?id=create_svc_rpc

image

ClientProcessId normally points to the PID of client process that requested the service creation via RPC (i.e. sc.exe or other utility, value here is equal to 0 since its a custom implementation and where high likely the RPC binary data  passed to the SCM were supplied with 0 which causes this abnormal values popylated for both ClientProcessId and its parent.

### CODE REVIEW

File: rules/windows/privilege_escalation_windows_service_via_unusual_client.toml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+administrator to SYSTEM.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-system.*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Windows Service Installed via an Unusual Client"
+note = """## Config
+
+`System Security Extension` audit subcategory need to be enabled to log event 4697 `A service was installed in the system`.
```
Comment (w0rk3r) [inline rules/windows/privilege_escalation_windows_service_via_unusual_client.toml:20]:
```suggestion
The 'Audit Security System Extension' logging policy must be configured for (Success)
Steps to implement the logging policy with with Advanced Audit Configuration:

Computer Configuration > 
Policies > 
Windows Settings > 
Security Settings > 
Advanced Audit Policies Configuration > 
Audit Policies > 
System > 
Audit Security System Extension (Success)
```

File: rules/windows/privilege_escalation_windows_service_via_unusual_client.toml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+"""
+references = [
+    "https://www.x86matthew.com/view_post?id=create_svc_rpc",
+    "https://docs.microsoft.com/en-us/windows/security/threat-protection/auditing/event-4697",
```
Comment (w0rk3r) [inline rules/windows/privilege_escalation_windows_service_via_unusual_client.toml:25]:
```suggestion
    "https://docs.microsoft.com/en-us/windows/security/threat-protection/auditing/event-4697",
    "https://github.com/atc-project/atomic-threat-coverage/blob/master/Atomic_Threat_Coverage/Logging_Policies/LP_0100_windows_audit_security_system_extension.md"
```

File: rules/windows/privilege_escalation_windows_service_via_unusual_client.toml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+]
+risk_score = 73
+rule_id = "55c2bf58-2a39-4c58-a384-c8b1978153c2"
+severity = "high"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Privilege Escalation"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.code:"4697" and (winlog.event_data.ClientProcessId:"0" or winlog.event_data.ParentProcessId:"0")
```
Comment (w0rk3r) [inline rules/windows/privilege_escalation_windows_service_via_unusual_client.toml:35]: Maybe add event.action to give a bit more context

**Review by w0rk3r — APPROVED** [review]
Nice 🔥

File: rules/windows/privilege_escalation_windows_service_via_unusual_client.toml (hunk 3 above)
Comment (Samirbous) [reply rules/windows/privilege_escalation_windows_service_via_unusual_client.toml:47]: good suggestion (done).

**Review by DefSecSentinel — APPROVED** [review]
LGTM

**Review by brokensound77 — APPROVED** [review]
Nice ✔️

File: etc/non-ecs-schema.json [hunk 4]
```diff
@@ -16,18 +16,20 @@
         "AttributeValue": "keyword",
         "CallerProcessName": "keyword", 
         "CallTrace": "keyword",
+        "ClientProcessId": "keyword",
         "GrantedAccess": "keyword",
+        "NewTargetUserName": "keyword",
         "ObjectDN": "keyword",
+        "OldTargetUserName": "keyword",
         "OriginalFileName": "keyword",
+        "ParentProcessId": "keyword"
```
Comment (brokensound77) [inline etc/non-ecs-schema.json:25]:
```suggestion
        "ParentProcessId": "keyword",
```

**Pull request merged**